tests/sentry/workflow_engine/processors/test_workflow.py

```diff
@@ -400,3 +408,75 @@ def test_enqueue_workflow__adds_to_workflow_engine_set(
+        self.workflow.save()
+
+        action_filter = self.create_data_condition_group()
+        action = self.create_action()
+        action_and_filter = self.create_data_condition_group_action(
+            condition_group=action_filter,
+            action=action,
+        )
+
+        workflow_actions = self.create_workflow_data_condition_group(
+            workflow=self.workflow, condition_group=action_filter
+        )
+
+        trigger_condition = self.create_data_condition(condition_group=workflow_trigger)
+        action_condition = self.create_data_condition(condition_group=action_filter)
+
+        self.workflow_id = self.workflow.id
+        self.workflow_trigger_id = workflow_trigger.id
+        self.action_filter_id = action_filter.id
+        self.action_id = action.id
+        self.action_and_filter_id = action_and_filter.id
+        self.workflow_actions_id = workflow_actions.id
+        self.trigger_condition_id = trigger_condition.id
+        self.action_condition_id = action_condition.id
+
+    # TODO - Figure out why parametrize isn't working for `test_delete_workflow`
+    def assert_clean_workflow_models(self):
+        assert not Workflow.objects.filter(id=self.workflow_id).exists()
+        assert not DataConditionGroup.objects.filter(id=self.workflow_trigger_id).exists()
+        assert not DataCondition.objects.filter(id=self.trigger_condition_id).exists()
+
+    def assert_clean_action_models(self):
+        assert not DataConditionGroup.objects.filter(id=self.action_filter_id).exists()
+        assert not Action.objects.filter(id=self.action_id).exists()
+        assert not DataConditionGroupAction.objects.filter(id=self.action_and_filter_id).exists()
+        assert not WorkflowDataConditionGroup.objects.filter(id=self.workflow_actions_id).exists()
+        assert not DataCondition.objects.filter(id=self.action_condition_id).exists()
+
+    def test_delete_workflow__deletes_workflow(self):
+        delete_workflow(self.workflow)
+        self.assert_clean_workflow_models()
+
+    def test_delete_workflow__deletes_action_tables(self):
+        delete_workflow(self.workflow)
+        self.assert_clean_action_models()
+
+    def test_delete_workflow__no_actions(self):
+        # Remove the assiaction action and reference to the action to the action conditions
+        Action.objects.get(id=self.action_id).delete()
+        assert not DataConditionGroupAction.objects.filter(id=self.action_and_filter_id).exists()
+
+        delete_workflow(self.workflow)
+        self.assert_clean_action_models()
+
+    def test_delete_workflow__no_workflow_triggers(self):
+        # TODO - when this condition group is deleted, it's removing the workflow
+        # it's basically inverted from what's expected on the cascade delete
+        self.workflow.when_condition_group = None
+        self.workflow.save()
+        DataConditionGroup.objects.get(id=self.workflow_trigger_id).delete()
```

Comment:
i noticed if i delete the `DataConditionGroup` it removes the workflow, but if the workflow is deleted the `Workflow.when_condition_group` isn't removed. 🤔 

Any idea why that would be the case? Here's where the [foreign key relationship is defined](https://github.com/getsentry/sentry/blob/master/src/sentry/workflow_engine/models/workflow.py#L33-L35)

Reply:
it's because the default behavior is cascade delete
https://github.com/getsentry/sentry/blob/ec3563d57678bc97305e317a9e784a747563790d/src/sentry/db/models/fields/foreignkey.py#L15
we'll need to change it to `models.SET_NULL`

Reply:
i don't think there's a reverse one to delete a FK field object when you delete a Django object

Reply:
yeah - i was just wondering if it was just me or if there's a way to accomplish the goal in django -- doesn't seem like it. (this shouldn't be a property on data condition group, because it'd be null for like 1/3 of them if we had the FK setup the other way).